src/Shared/Cosmos/CosmosUtils.cs

```diff
@@ -23,4 +24,64 @@ internal static bool IsEmulatorConnectionString(string? connectionString)
         var accountKeyFromConnectionString = v.ToString();
         return accountKeyFromConnectionString == CosmosConstants.EmulatorAccountKey;
     }
+
+    /// <summary>
+    /// Parses the connection string to extract the account endpoint and connection string.
+    /// </summary>
+    /// <remarks>
+    /// The connection string can be in the following formats:
+    /// A valid Uri
+    /// AccountEndpoint={valid Uri} - optionally with [;Database={databaseName}[;Container={containerName}]]
+    /// {valid CosmosDB ConnectionString} - optionally with [;Database={databaseName}[;Container={containerName}]]
+    /// </remarks>
+    internal static CosmosConnectionInfo ParseConnectionString(string connectionString)
+    {
+        Uri? accountEndpoint;
+        if (Uri.TryCreate(connectionString, UriKind.Absolute, out accountEndpoint))
+        {
+            return new CosmosConnectionInfo(accountEndpoint, null);
+        }
+
+        var connectionBuilder = new DbConnectionStringBuilder()
```

Comment:
future: If we're going all in on connection strings, it might be worth shipping a public API that uses `DbConnectionStringBuilder` internally.

Reply:
e.g. https://github.com/dotnet/aspire/issues/7427

Reply:
nit: `CosmosConnectionInfo` and `CosmosConnectionInfo.ParseConnectionString`

Reply:
Agreed. Sorry for the confusion, I was using the existing internal names above. Not what I would use for public API.